File: library/src/main/java/com/hansoolabs/and/view/RecyclerViewCalculator.java
package com.hansoolabs.and.view;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;

/**
 * Get position from scrollProgress or
 * get scrollProgress from position of RecyclerView item
 * Created by brownsoo on 2017. 8. 23..
 */

public interface RecyclerViewCalculator {
    int calculatePositionFromScrollProgress(
            @NonNull RecyclerView recyclerView,
            float progress);
    float calculateScrollProgress(
            @NonNull RecyclerView recyclerView);
}

File: library/src/main/kotlin/com/hansoolabs/and/view/RecyclerViewCalculator.kt
package com.hansoolabs.and.view

import android.support.v7.widget.RecyclerView

/**
 * Get position from scrollProgress or
 * get scrollProgress from position of RecyclerView item
 * Created by brownsoo on 2017. 8. 23..
 */

interface RecyclerViewCalculator {
    fun calculatePositionFromScrollProgress(
            recyclerView: RecyclerView,
            progress: Float): Int

    fun calculateScrollProgress(
            recyclerView: RecyclerView): Float
}
